modules/openapi-generator/src/main/resources/csharp/libraries/generichost/api.mustache

```diff
@@ -394,14 +394,9 @@ namespace {{packageName}}.{{apiPackage}}
                     {{#required}}
                     uriBuilderLocalVar.Path = uriBuilderLocalVar.Path.Replace("%7B{{baseName}}%7D", Uri.EscapeDataString({{paramName}}.ToString()));
                     {{/required}}
-                    {{^required}}
```

Comment:
If we are making this change, then I think lines 394 and 396 should also go away. Perhaps replace it with a comment that all pathParams are required per openapi spec.

Reply:
done via https://github.com/OpenAPITools/openapi-generator/pull/17525/commits/aa64bd59cf83dd50a76adeb191adb7746b074b0f